watch/sinks_test.go

```diff
@@ -0,0 +1,43 @@
+package watch
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+// TestTimeoutDropErrSinkGen tests the full chain of sinks
+func TestTimeoutDropErrSinkGen(t *testing.T) {
+	require := require.New(t)
+
+	sinkGen := NewTimeoutDropErrSinkGen(time.Second)
+
+	// Generate two channels to perform the following test-cases
+	sink, ch := sinkGen.NewChannelSink()
+	sink2, ch2 := sinkGen.NewChannelSink()
+
+	go func() {
+		for {
+			<-ch.C
+		}
+	}()
```

Comment:
Looks like this test will leak a goroutine; is `ch.C` ever closed? If not, can you add another channel for this goroutine to `select` on so it will terminate when the test is finished?